src/cli/migrate/index.ts

```diff
@@ -0,0 +1,54 @@
+import { log } from '../../utils/logging';
+
+import { CURRENT_NODE_LTS, nodeVersionMigration } from './nodeVersion';
+
+const migrations = {
+  'node-version': {
+    migrate: nodeVersionMigration,
+    getArgs: (args: string[]): [number] => {
+      if (args.length === 0) return [CURRENT_NODE_LTS];
+
+      const version = Number(args[0]);
+      if (Number.isNaN(version)) {
+        log.err('Provide a valid Node.js version to migrate to.');
+        process.exit(1);
+      }
+
+      return [version];
+    },
+  },
+};
+
+type Migration = keyof typeof migrations;
+
+const logAvailableMigrations = () => {
+  log.ok('Available migrations:');
+  Object.keys(migrations).forEach((migration) => {
+    log.ok(`- ${migration}`);
+  });
+};
+
+export const migrate = async (args = process.argv.slice(2)) => {
+  if (!args[0]) {
+    log.err('Provide a migration to run.');
+    logAvailableMigrations();
+    process.exitCode = 1;
+    return;
+  }
+
+  if (args.includes('--help') || args.includes('-h') || args[0] === 'help') {
+    logAvailableMigrations();
+    return;
+  }
+
+  if (!migrations[args[0] as Migration]) {
```

Comment:
Is the cast needed if we're checking for falsy values above?

Reply:
it was because the type was { 'node-version': stuff } rather than Record<string, stuff> but fixed now with other comment